servers/cu/src/domain/client/metrics.js

```diff
@@ -48,13 +74,14 @@ export const gaugeWith = ({ prefix = 'ao_cu' } = {}) => {
         ? { collect: async function () { this.set(await collect()) } }
         : {}
       ),
+      labelNames,
       enableExemplars: true
     })
 
     return {
-      inc: (n) => g.inc(n),
-      dec: (n) => g.dec(n),
-      set: (n) => g.set(n)
+      inc: (n, labels) => labels ? g.labels(labels).inc(n) : g.inc(n),
+      dec: (n, labels) => labels ? g.labels(labels).dec(n) : g.dec(n),
+      set: (n, labels) => labels ? g.labels(labels).set(n) : g.set(n)
```

Comment:
`inc`, `dec` and `set` can take an object that contains `{ value, labels, exemplars }` as well. So you might not need to expand the api surface.

Reply:
This is only true for counters, not for gauges